src/jit/morph.cpp

```diff
@@ -12422,6 +12424,25 @@ GenTree* Compiler::fgMorphSmpOp(GenTree* tree, MorphAddrContext* mac)
                 }
             }
 
+            // Transform X MOD C == 0 to X UMOD C == 0 if C is a power of two
+            // then lowering will be able to transform it to X & (C-1) == 0
+            if (op2->IsIntegralConst(0))
+            {
+                op1 = tree->AsOp()->gtGetOp1();
+                if (op1->OperIs(GT_MOD))
+                {
+                    if (varTypeIsIntegralOrI(op1->AsOp()->gtGetOp1()->gtType) &&
+                        (op1->AsOp()->gtGetOp2()->IsIntegralConst()))
+                    {
+                        ssize_t divider = op1->AsOp()->gtGetOp2()->AsIntCon()->IconValue();
+                        if (isPow2(divider))
+                        {
+                            op1->SetOper(GT_UMOD);
```

Comment:
I'm not familiar with the JIT, but, shouldn't `divider` be updated somewhere to be `divider - 1`?